gc3libs/config.py

```diff
@@ -401,6 +400,16 @@ def _parse(self, stream, filename=None):
                                 filename=filename
                             ))
 
+                for key in self.resource_key_value_matcher.keys():
+                    if key in config_items and re.compile(self.resource_key_value_matcher[key][0]).match(config_items[key]):
```

Comment:
This is compiling the same RE over and over again, potentially. While not a big waste of CPU cycles, you can just compile the regex when defining `resource_key_value_matcher` below.